Add Ayon version control addon docs (#242)
ynput/ayon-documentation: website/docs/addon_version_control_admin_get_started.md

---
id: addon_version_control_admin_get_started
title: Version Control Get Started
sidebar_label: Get Started
description: Version Control Get Started.
toc_max_heading_level: 5
---

import ReactMarkdown from "react-markdown";
import versions from '@site/docs/assets/json/Ayon_addons_version.json'

<ReactMarkdown>
    {versions.VersionControl_Badge}
</ReactMarkdown>

## Current state of the addon

This addon tries to implement generic API for various version control system.

Currently contains WIP implementation of Perforce, but there might be more in the future which should follow
same API methods.

> Implementation tries to use only single dependency `p4python` which is binary dependent on version of python used.

Reviewer: does it also mean like this?
```suggestion
Implementation tries to use only single dependency `p4python` which is dependent on binary version of python used.
```

> Which might be different in different DCCs, different versions of Unreal etc.

Reviewer: ```suggestion
This means it might be different in different DCCs, different versions of Unreal etc.
```

To mitigate need of having binary compatible `p4python` libraries addon currently implements REST api to run p4 commands 
only on separate webserver started by AYON Tray which contains `p4python` library installed via dependency package and 
REST stub class which might be used in each DCC that has `requests` library.

:::info
The AYON Version Control addon doesn't replace the built-in versioning functionality.
Instead, it adds an extra layer by publishing the product to two separate locations: the AYON publish directory and the Perforce server.
:::

## Configuration of connection to Perforce server

Addons offers configuration of credentials to P4 server with username and password on multiple levels via
Studio(single credentials), Project (credentials different per project)  or Site settings(credentials different per artist.)

## Perforce workspace

> It is expected that P4 workspaces would be setup and existing on artists machines. Each artist then only 

Reviewer: ```suggestion
It is expected that P4 workspaces would be set up and existed on artists machines. Each artist then only 
```

Author: https://github.com/ynput/ayon-documentation/pull/242/commits/3a2777db85f45154e2cf09ef3a60f3e668b618ea

provides path to their locally accessible P4 workspace folder in Site Settings.

## Main functionalities for Perforce

> This addons contains two main  functionality for slightly different use cases:

Reviewer: ```suggestion
This addons contains two main functionalities for slightly different use cases:
```

> This addons contains two main  functionalities for slightly different use cases:

Reviewer: double space
```suggestion
This addons contains two main functionalities for slightly different use cases:
```

- create separate version controlled streams from AYON of published products in Perforce
- help with Deadline rendering of Unreal projects with Perforce

> It is expected that majority work for Perforce with Unreal project files are done with official Perforce

Reviewer: ```suggestion
It is expected that majority working for Perforce with Unreal project files are done with official Perforce
```

tools like `P4V` or command line utility. Neither of those are distributed by addon and it is required
for IT department to install/configure them.

### Commits to Perforce

This approach allows to copy version control not only in AYON, where each publish results in new version
of published product, but committing this version directly to Perforce. 

> That way versions of particular product are separately kept in Perforce, from which ordinary Perforce tools

Reviewer: It could be like....
```suggestion
The products with versions are separately kept in Perforce, from which ordinary Perforce tools
```

Author: [tweak some words](https://github.com/ynput/ayon-documentation/pull/242/commits/3a2777db85f45154e2cf09ef3a60f3e668b618ea)

could be used (P4V) to pull latest version of published product.

Profiles could be used to limit which published products should be committed to Perforce.

> As Perforce controlled product doesn't contain version name directly in its name nor path, additional

Reviewer: ```suggestion
As Perforce controls the product which doesn't contain version name directly in its name nor path, additional
```
